app/controllers/courses_controller.rb

```diff
@@ -17,6 +17,14 @@ def index
   # GET /courses/1
   # GET /courses/1.json
   def show
+    session[:course_id] = @course.id
```

Comment:
I tried doing that several different ways and couldn't get it to work. I ended up settling on the session one because I got it to work. I can take another stab at it.